### DISCUSSION TITLE
オートコンプリート機能にfuzzy検索機能を加える

### DISCUSSION THREAD

**Discussion by blueowlgreenfish** [pgroonga/pgroonga#244] [Q&A]
[オートコンプリート機能の実装方法](https://pgroonga.github.io/ja/how-to/auto-complete.html)にしたがって、`terms`のテーブルを作成し、2つのインデックスを加えて、`INSERT INTO terms (term, readings) VALUES ('牛乳', ARRAY['ギュウニュウ', 'ミルク']);`を実行した後で、`SELECT term FROM terms WHERE readings &^~ 'ぎゅう';`を実行すると、`牛乳`がしっかりヒットします。このオートコンプリートの機能を持ったカラム`readings`に、fuzzy検索機能を加えることはできますでしょうか？例えば、`SELECT term FROM terms WHERE readings &^~ 'ぎゅあ';`を実行すると、牛乳がヒットするようにしたいです。fuzzy検索は、`` &` ``の演算子を使って、`fuzzy_search`や`pgroonga_escape`や`with_transposition`や`max_distance`といったキーワードを使うと思いますが、オートコンプリート機能にfuzzy検索機能をどうすれば加えることができますでしょうか？

**Comment by komainu8** [comment]
手元で色々試してみたのですが、既存の機能ですんなりとはできなさそうです。
回答まで少々お待ちください。
